### DISCUSSION TITLE
質問： 大量のデータを localStorage に書き込む必要性を教えてください

### DISCUSSION THREAD

**Discussion by bindi-mo** [increments/qiita-discussions#570] [要望・アイデア・不具合]
Firefox の Profile を SSD から HDD に移動させてみたら、HDDがゴリゴリと非常に耳障りな状態になったので、調べてみました。

Qitta のウェブサイトページを開いていると、数キロバイトを複数個所（？）にわたり、書き込んでいました。
しばらくしたら止まるかと思いきや、永遠に書き込んでいます。
SDD の寿命を短くするためにやっているのでしょうか？（苦笑）

なぜ、そんなに書き込む必要があるのでしょうか？

参考
\* [localStorage を使ってみる](https://qiita.com/masuda-sankosc/items/cff6131efd6e1b5138e6)
\* [Window: localStorage プロパティ](https://developer.mozilla.org/ja/docs/Web/API/Window/localStorage)

**Comment by shiracamus** [comment]
私も、Qiitaページを開くとPC本体のHDDアクセスランプが1秒に3回くらい点滅を続けるのがずっと気になってました。
SSD寿命が気になります。

**Comment by tanaka-asami415** [comment]
@AquaBindi @shiracamus 
ご報告ありがとうございます！
一度調査いたします🙇

**Reply by bindi-mo** [reply]
## 解決したかも？
いつのまにか書き込みをしなくなりました。
やったことは下記の通り

- HOME (https://qiita.com) にいって、そこにあるトピックスの記事をみる。 （3件ぐらい見ました）
- タイムライン (https://qiita.com/timeline) を表示。
- トレンド (https://qiita.com/trend) を表示。
- マイページにいく。

などをしていたら、いつのまにか書き込みをしなくなりました。

qiita.com のキャッシュや localStorage をすべて削除して、 任意の記事だけを表示。
それで再現できればよいのですが。

ちなみにわたしは、 Windows 11 上のリソースモニターで表示される Disk をみて、この状態を確認しました。

## 余談
アカウントを持っていて、コメントや記事を数件投稿はしているのですが、もっぱら Google 検索の結果から qiita.com に行く。という利用をしています。HOME から利用することはありません。

## 蛇足
qiita.com 上の下書き多数、ローカルのパソコンにも下書き記事が多数。書いている途中で、すぐに飽きてしまいます(笑)

**Reply by bindi-mo** [reply]
再発しました

トレンドを開きっぱなしのタブがあり、記事の投稿を新しくタブで開く。
その状態でしばらく放置していたらトレンドのほうで、がりがり書き込みが発生。
記事の投稿では発生しません。

あとは、ウェブブラウザが Firefox だからかな？
Microsoft Edge, Google Chrome で、 qitta.com にログインせずに記事をみたかぎり、 localStorage に書き込んでいる様子はない。

原因を調べるのは大変そうです。

**Reply by bindi-mo** [reply]
news.mynavi.jp さんで、同じ症状になりました。
広告の Javascript がやっているのではないでしょうか

**Reply by tanaka-asami415** [reply]
追加の情報をありがとうございます🙇
調査の参考にさせていただきます！

**Comment by mimaburao** [comment]
私も同様にqiitaのサイトに来訪するとHDDアクセスが止まらなくなりました。
chronium系のブラウザで一応低減か止めることができた。

原因は私の場合はjavascriptメモリーが増え続けることみたいでした。

特定方法として、chronuim系(私はbrave,Thoriumでやってみました)で、「その他のツール」→「タスクマネージャー」においてqiitaのタスクを参照します。右クリックメニューでJavascripメモリもオンにして様子をみると、数値がどんどんと上がっていきます。

解決として、javascripをqiita上で無効にすることでHDDアクセスが止んできました。
ブラウザの設定(braveの場合)において、brave://settings/content/javascriptにアクセスすると動作のカスタマイズ欄のjavascriptの使用を許可しないにqiita.comを入れる。

**Reply by tanaka-asami415** [reply]
@mimaburao 
ご報告と詳細を教えていただきありがとうございます！
現在調査中ですので、こちらも参考にさせていただきます🙇

**Comment by ohakutsu** [comment]
@AquaBindi @shiracamus @mimaburao 
おまたせしました！
お問い合わせいただいたものとは少し異なるのですが、特定の環境にてindexedDBに書き込みが行われ続けることが確認でき、修正をしました！
もし書き込みが行われ続けている様でしたら、またご報告いただけると幸いです。

**Reply by mimaburao** [reply]
braveブラウザーにてqiitaサイト上でjavascripの使用をできるようにしてみました。javascripメモリの継続的な上昇はみられず、HDDアクセスも以前のとは異なっているようです。継続して使用してみます。特定と対応お疲れ様です。
